src/main/java/core/db/User.java

```diff
@@ -1,138 +1,146 @@
-// %1916127255:de.hattrickorganizer.model%
 package core.db;
 
+import core.HO;
 import core.util.HOLogger;
+import core.util.OSUtils;
 import tool.updater.UpdateHelper;
-
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.OutputStreamWriter;
-import java.io.PrintWriter;
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
-
 import org.w3c.dom.Document;
 import org.w3c.dom.Element;
 import org.w3c.dom.NodeList;
 import org.w3c.dom.Text;
 
 public class User {
-
+	// Class variables
 	private static List<User> users = null;
-	private static final String DEFAULT_FILE = "user.xml.default";
 	private static final String FILENAME = "user.xml";
 	private static final String ENCODING = "UTF-8";
-	public static int INDEX = 0;
-	private String driver = "org.hsqldb.jdbcDriver";
-	private String name = "singleUser";
-	private String pwd = "";
-	private String url = "jdbc:hsqldb:file:db/database";
+	private static String driver = "org.hsqldb.jdbcDriver";
+	private static String dbParentFolder;
+
+	private String dbName;
+	private String dbURL;
 	private String user = "sa";
 	private int backupLevel = 3;
 	private boolean isNtTeam = false;
+	private String dbFolder;
+	public static int INDEX = 0;
+	private String name = "singleUser";
+	private String pwd = "";
 
-	/**
-	 * Creates a new User object.
-	 */
-	private User() {
+
+    // getters and setters
+	public String getDbName() {return dbName;}
+	public void setDbName(String dbName) {this.dbName = dbName;	}
+	public final String getDriver() {return driver;	}
+	public String getDbFolder() {return dbFolder;}
+	public final String getTeamName() {return name;}
+	public final String getPwd() {return pwd;}
+	public final String getUser() {return user;}
+	public int getBackupLevel() {return backupLevel;}
+	public boolean isNtTeam() {return isNtTeam;}
+	public final void setName(String name) {this.name = name;}
+	public final void setBackupLevel(int level) {backupLevel = level;}
+	public boolean setNtTeam(boolean isNtTeam) {return this.isNtTeam = isNtTeam;}
+
+	public void setURL() {
+		if (HO.isPortableVersion()) dbURL = "jdbc:hsqldb:file:" + dbName + "/database";
+		else dbURL = "jdbc:hsqldb:file:" + dbFolder + "/database";
 	}
 
-	public static List<User> getAllUser() {
-		try {
-			if (users == null) {
-				load();
+	public static String getDbParentFolder() {return dbParentFolder;}
+
+	private static void setDBParentFolder(){
+		if (! HO.isPortableVersion())
+		{
+			if (HO.getPlatform() == OSUtils.OS.LINUX) {
+				dbParentFolder = System.getProperty("user.home") + "/.ho";}
+			else if (HO.getPlatform() == OSUtils.OS.MAC) {
+				dbParentFolder =  System.getProperty("user.home") + "/Library/ApplicationSupport/HO";}
```

Comment:
There should be a space between `Application` and `Support`.

Reply:
I think it was correct before so I don't know how I end up changing this one ...

Reply:
> Apologies, I had just started the review when you merged.

No worry, I merged the PR because it started to be big and rebasing to master made the PR really messy. We might still have some issue but I think we are not far to have something solid

Reply:
done in 9c8ddeeab3cce923541ef88f0397bae914caa0cc